### ISSUE TITLE
LSTM Error when checking target: expected to have 2 dimensions, but got array with 3D shape

### ISSUE THREAD

**Issue by mejdidallel** [keras-team/keras#13573]
I am working on an LSTM project. 

Let's I have 100 input data. My input layer uses a slinding windows equal to 10.

Here's my LSTM model : 

    inputs = tf.keras.Input(shape=(10,100))
    LSTM_1 = layers.LSTM(100, stateful=False, return_sequences=True)(inputs)
    FC_1 = layers.Dense(100)(LSTM_1)
    LSTM_2 = layers.LSTM(100, stateful=False, return_sequences=False)(FC_1)
    label = layers.Dense(5, activation='softmax')(LSTM_2)

Here's the model summary.

    Layer (type)                 Output Shape              Param #   
    =================================================================
    input (InputLayer)        (None, 10, 100)           0         
    _________________________________________________________________
    lstm_1 (LSTM)             (None, 10, 100)           80400     
    _________________________________________________________________
    dense (Dense)             (None, 10, 100)           10100     
    _________________________________________________________________
    lstm_2 (LSTM)             (None, 100)               80400     
    _________________________________________________________________
    label (Dense)             (None, 5)                1111      
    =================================================================
    Total params: 172,011
    Trainable params: 172,011
    Non-trainable params: 0
    _________________________________________________________________
    None

Below is the code to load data.

    h5_x = 'X.h5'
    h5_Y = 'Y.h5' 
    
    x_data = HDF5Matrix(h5_x, start=0, end=data_length)
    x_data = np.reshape(x_data,(data_length, 5))       
    
    y_label = HDF5Matrix(h5_Y, start=0, end=data_length) 
    y_label = np.reshape(y_label,(data_length, 5))

And let's say I have a function that splits data to sequences of 10 and feed them to the model.

My question is when I set ***return_sequences*** to ***True***, my code works fine feeding the model with sequences of 10 slices but the output layer is of a shape **(None, 10, 5)** which I don't want it this way. 

So when I set it to **False**, the output layer is of a shape **(None, 5)**, but I get the error message : Error when checking target: expected to have 2 dimensions, but got array with shape (x, x, x). 

I know it's just a shape problem but how can I solve this ? What do I need to reshape ?
